src/amber.cc

```diff
@@ -171,7 +172,8 @@ amber::Result Amber::ExecuteWithShaderData(const amber::Recipe* recipe,
   Result executor_result = executor.Execute(
       engine.get(), script, opts->delegate, shader_data,
       opts->pipeline_create_only ? ExecutionType::kPipelineCreateOnly
```

Comment:
I think we should just pass options into `Execute` here as this is getting long ....

Reply:
I agree. I have done this. However, `Options` includes:

```c++
EngineType engine;
std::string spv_env;
std::vector<BufferInfo> extractions;
```

none of which are used in `Execute`. I guess that is OK though, although it could be a bit confusing if future code ends up trying to rely on these fields.

I also need to update about 29 tests which currently do this:

```c++
Executor ex;
Result r = ex.Execute(engine.get(), script.get(), nullptr, ShaderMap(),
                        ExecutionType::kExecute);
```

to something like this:

```c++
Options options;
Executor ex;
Result r = ex.Execute(engine.get(), script.get(), ShaderMap(), &options);
```

`options->engine` will be `amber::EngineType::kEngineTypeVulkan` but the actual engine is `EngineStub`. But again, this works, as `Execute` does not access `options->engine`. Could be a bit confusing though.